### PR TITLE
Deprecate rmdir, mkdir_with_final_link, prune! and prune_ancestors

### LINKED ISSUE #16: Deprecate. DruidTools#rmdir, #prune! and #prune_ancestors
See https://github.com/sul-dlss/dor-services-app/issues/326
and https://github.com/sul-dlss/dor-services-app/pull/328/files
Other than the usage in dor-services-app, these methods are not used.

### PR DESCRIPTION
**Pull request by jcoyne**
Fixes #16

### CODE REVIEW

File: lib/druid_tools/druid.rb [hunk 1]
```diff
@@ -122,6 +126,7 @@ def mkdir_with_final_link(source, extra = nil)
       FileUtils.mkdir_p(real_path)
       FileUtils.ln_s(source, new_path, force: true)
     end
+    deprecation_deprecate :mkdir_with_final_link
```
Comment (mjgiarlo) [inline lib/druid_tools/druid.rb]: Should these include some text about what to use instead?

Comment (jcoyne) [reply lib/druid_tools/druid.rb]: They aren't being replaced.

File: Gemfile [hunk 2]
```diff
@@ -3,11 +3,7 @@
 source 'https://rubygems.org'
 
 group :development do
-  gem 'debugger', platform: :ruby_19
-  gem 'pry-debugger', platform: :ruby_19
-  gem 'rcov', platform: :ruby_18
-  gem 'ruby-debug', platform: :ruby_18
-  gem 'simplecov', platform: %i[ruby_19 ruby_20]
```
Comment (ndushay) [inline Gemfile]: do we want to cease code coverage?

Comment (jcoyne) [reply Gemfile]: This was for platform ruby_20 and 19, but we aren't using those rubies.

Comment (ndushay) [reply Gemfile]: thanks.

**Pull request merged**